src/printer.js

```diff
@@ -3880,7 +3880,29 @@ function makeString(rawContent, enclosingQuote) {
     return match;
```

Comment:
Isn't it enough to simply change the above lines to this and not do anything else?

```js
if (quote) {
  return quote;
}

return /^[^\\nrvtbfux\r\n\u2028\u2029"'0-7]$/.test(escaped)
  ? escaped
  : '\\' + escaped
```